components/enedis/enedis.app.mjs

```diff
@@ -0,0 +1,123 @@
+import { axios } from "@pipedream/platform";
+
+export default {
+  type: "app",
+  app: "enedis",
+  propDefinitions: {
+    start: {
+      label: "Start",
+      description: "Start date of the period in RFC-3339 format (YYYY-MM-DD) e.g. '2022-10-28'. The returned datas will include the start date.",
+      type: "string",
+    },
+    end: {
+      label: "End",
+      description: "End date of the period in RFC-3339 format (YYYY-MM-DD) e.g. '2022-11-01'. The returned datas will not include the end date.",
+      type: "string",
+    },
+    usage_point_id: {
+      label: "Usage point id",
+      description: "Usage point id known by the user. Use '12655648759651' if don't have one.",
+      type: "string",
+    },
+  },
+  methods: {
+    _getUrl(path) {
+      return `https://ext.hml.api.enedis.fr${path}`;
+    },
+    _getHeaders() {
+      return {
+        "Authorization": `Bearer ${this.$auth.oauth_access_token}`,
+        "Accept": "application/json",
+        "Content-Type": "application/json",
+		"User-Agent": "@PipedreamHQ/pipedream v0.1",
+      };
+	},
+    prepareParam(self) {
+      return {
+        "usage_point_id": self.usage_point_id,
+      };
+    },
+    prepareAllParams(self) {
+      return {
+        "usage_point_id": self.usage_point_id,
+        "start": self.start,
+        "end": self.end,
+      };
+    },
```

Comment:
These methods should be in a common action - so that actions inherit them and are able to call them. Preparing the params is the action component's responsibility - the app should receive these params ready to make the API call.

[The Pipedream docs describe this and show a good example app](https://pipedream.com/docs/components/guidelines/#common-files-optional). Here's a short example of how it'd go about for this component:

You could have a `common` action file that defines these methods, and the other actions `import` that common code directly. The common code would be something like:

```js
export default {
  methods: {
    prepareParam() {
      return {
        "usage_point_id": this.usagePointId
      }
    }
  }
}
```
And the other actions would import it:
```js
import common from "../common.mjs"
export default {
  ...common,
  // individual action code comes after the common code
}
```

This way, you do not need to pass the entire action (`this`) to the app - since the method is in the action's scope, you can simply access the props using `this` as you would in the `run` method. The app then receives only the params, as it should.